core/ledger/snapshotgrpc/snapshot_service.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package snapshotgrpc
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/empty"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/pkg/errors"
+)
+
+// Snapshot Service implements SnapshotServer grpc interface
+type SnapshotService struct {
+	LedgerGetter LedgerGetter
+}
+
+// LedgerGetter gets the PeerLedger associated with a channel.
+type LedgerGetter interface {
+	GetLedger(cid string) ledger.PeerLedger
+}
```

Comment:
Let's avoid these interfaces. We can directly use the object, i.e., `LedgerMgr`.